filterdb/db.go

```diff
@@ -163,26 +163,38 @@ func putFilter(bucket walletdb.ReadWriteBucket, hash *chainhash.Hash,
 	return bucket.Put(hash[:], bytes)
 }
 
-// PutFilter stores a filter with the given hash and type to persistent
-// storage.
+// PutFilters stores a set of filters to persistent storage.
 //
 // NOTE: This method is a part of the FilterDatabase interface.
-func (f *FilterStore) PutFilter(filterData *FilterData) error {
-	return walletdb.Update(f.db, func(tx walletdb.ReadWriteTx) error {
+func (f *FilterStore) PutFilters(filterList ...*FilterData) error {
+	return walletdb.Batch(f.db, func(tx walletdb.ReadWriteTx) error {
 		filters := tx.ReadWriteBucket(filterBucket)
 
-		var targetBucket walletdb.ReadWriteBucket
-		switch filterData.Type {
-		case RegularFilter:
-			targetBucket = filters.NestedReadWriteBucket(regBucket)
-		default:
-			return fmt.Errorf("unknown filter type: %v",
-				filterData.Type)
+		for _, filterData := range filterList {
+			var targetBucket walletdb.ReadWriteBucket
+			switch filterData.Type {
+			case RegularFilter:
+				targetBucket = filters.NestedReadWriteBucket(
```

Comment:
nit: instead of calling `NestedReadWriteBucket` in every iteration we can call it once before the loop and change targetBucket in the switch statement. Now we have only two optio: regBucket or error but this would work even if we add more in the future